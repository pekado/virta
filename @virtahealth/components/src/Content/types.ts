// eslint-disable-next-line @typescript-eslint/no-namespace
export namespace ContentTypes {
  interface ContentBase {
    _id: string;
    _i18NLang: string;
    careProtocols: Array<any>;
    heroImage: HeroImage;
    previewText: string;
    slug: Slug;
    tags: Array<Tag>;
  }

  export interface ArticleType extends ContentBase {
    body: Array<any>;
    title: string;
    _type: "article";
  }

  export interface Tag {
    tagName: string;
    tagSrc: string;
  }

  export interface Slug {
    current: string;
  }

  export interface HeroImage {
    altText: string;
    caption: string;
    imageUrl: string;
  }

  export interface RecipeType extends ContentBase {
    body: Array<any>;
    name: string;
    ingredients: RecipeIngredient[];
    nutritionFacts: {
      calorieCount: number;
      carbohydrates: Quantity;
      fat: Quantity;
      fiber: Quantity;
      protein: Quantity;
      sugarAlcohol?: Quantity;
      sugarAdded?: Quantity;
      servingSize: Quantity;
      servingCount: number;
    };
    nutritionNotes?: string;
    preface: Array<any>;
    prepTime: Quantity;
    totalTime: Quantity;
    instructions: string[];
    _type: "recipe";
  }

  export interface RecipeIngredient {
    ingredient: string;
    notes?: string;
    quantity: Quantity;
  }

  export interface Quantity {
    amount: number;
    unit: {
      abbreviation?: string;
      label: string;
      labelPlural?: string;
    };
  }
}
