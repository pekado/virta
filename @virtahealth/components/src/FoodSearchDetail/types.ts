export namespace FoodDetailTypes {
  export interface Quantity {
    amount: number;
    unit: {
      abbreviation?: string;
      label: string;
      labelPlural?: string;
    };
  }
  export interface NutritionalInformation {
    protein: Quantity;
    fat: Quantity;
    carbohydrate: Quantity;
    calories: number;
    servingSize: Quantity;
  }
  export interface BaseFoodData {
    id?: string;
    name: string;
    picture: string;
    description: string;
  }
  export interface BrandData {
    brandId: string;
    brandName: string;
    location: string;
    price: number;
  }
  export interface FoodItem {
    baseFoodData: BaseFoodData;
    nutritionalInformation: NutritionalInformation;
    brandData?: BrandData;
  }
}
