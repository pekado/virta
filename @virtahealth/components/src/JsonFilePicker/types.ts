import { defineMessages, MessageDescriptor } from "react-intl";

export enum AcceptedFileTypes {
  pdf = ".pdf",
  jpg = ".jpg",
  jpeg = ".jpeg",
  png = ".png",
  gif = ".gif",
}

export interface UploaderProps {
  accept: AcceptedFileTypes[];
  onSetFiles: (files: File[]) => void;
  labelMessage?: MessageDescriptor;
  testID?: string;
}

export const labelMessage = defineMessages({
  filePickerLabel: {
    id: "filePicker.defaultLabel",
    description: "The default file drop label",
    defaultMessage: "Select File",
  },
});

export const defaultProps = {
  labelMessage: labelMessage.filePickerLabel,
};
