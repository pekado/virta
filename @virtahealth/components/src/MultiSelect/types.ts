export interface ListItem {
  id: number;
  labelMessage: string;
}

export interface SelectedListItems {
  [itemId: number]: ListItem;
}
