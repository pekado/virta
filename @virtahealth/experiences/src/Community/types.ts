export interface CommunityUser {
  id: number;
  days_on_virta: number;
  display_name: string;
  is_active: boolean;
  profile_picture_url: string;
  role: string;
  posts_and_comments?: Array<CommunityPost | CommunityComment>;
  virta_user_id?: number;
  tos_accepted_on?: string;
}

export interface CommunityPost {
  id: number;
  topic_id: number;
  body: string;
  media: string[];
  comment_count: number;
  comments: CommunityComment["id"][];
  created_at: string;
  like_count: number;
  liked: boolean;
  author: CommunityUser["id"];
}

export interface CommunityComment {
  id: number;
  post_id: number;
  author: CommunityUser["id"];
  body: string;
  media: string[];
  liked: boolean;
  like_count: number;
  created_at: string;
}

export interface CommunityTopic {
  id: number;
  name: string;
  is_small_group: boolean;
  position: number;
  // TODO: Is this needed? Added categoryId in as a hack to fix broken build, but it isn't populated anywhere
  // This field is referred to in TopicsPicker.
  categoryId: number;
}

export interface CommunityTopicByCategoryId {
  [categoryId: number]: CommunityTopic;
}

export interface CreatePostPayload {
  media: any;
  body: string;
  category_id?: number;
}

export interface UpdatePostPayload {
  post_id: number;
  category_id?: number; // This is CommunityTopic.id
  media: any;
  body: string;
}

export interface CreateCommentPayload {
  topic_id: number; // This is CommunityPost.id
  body: string;
  media: any;
}

export interface UpdateCommentPayload {
  topic_id: number; // This is CommunityPost.id
  post_id: number; // This is CommunityComment.id
  media: any;
  body: string;
}

export interface UpdateCommunityUserPayload {
  display_name: string;
  use_default_profile_picture: boolean;
  new_photo_key: string; // I think?
}
