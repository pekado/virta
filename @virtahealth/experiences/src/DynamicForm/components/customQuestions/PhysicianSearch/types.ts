import { CamelCaseEnrollmentAddress } from "../../../../types";

export interface SearchResultsJSON {
  data: SearchResult[];
  total: number;
}
export interface SearchResult {
  address: CamelCaseEnrollmentAddress;
  firstName: string;
  lastName: string;
  phoneNumber: string;
}
export type SearchResultsByKey = Record<string, SearchResult[]>;
