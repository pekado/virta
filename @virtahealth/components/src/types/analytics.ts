export interface CommonAnalyticsProperties {
  screen_name: string;
  screen_path: string;
  destination_path?: string;
}

interface SkillTileProperties {
  skill_title: string;
}

interface BehavioralHealthScreenerCommonProperties
  extends CommonAnalyticsProperties {
  patient_action_id: string;
}
interface BehavioralHealthQuestionProperties
  extends BehavioralHealthScreenerCommonProperties {
  screener_question_display: string;
  screener_answer_display: string;
}
interface BehavioralHealthFollowUpEventProperties
  extends BehavioralHealthScreenerCommonProperties {
  skill_name: string;
}
interface ContentPageProperties extends CommonAnalyticsProperties {
  origin?: string;
  /**
   * `rank` represents the position within the page or carousel
   * of the content piece
   *
   * value should start at 1 (first piece of content is a "1")
   */
  rank: number;
}
export interface DiscoverSearchProperties {
  searchTerm: string;
  numberOfHits?: number;
}
type DiscoverSearchPageProperties = ContentPageProperties &
  DiscoverSearchProperties;

interface ActionCardAnalyticsProperties extends CommonAnalyticsProperties {
  content_id?: string;
  action_id?: string;
  action_title?: string;
}

type SkillsHomeEventProperties = CommonAnalyticsProperties &
  SkillTileProperties;
type BehavioralHealthEventProperties =
  | BehavioralHealthScreenerCommonProperties
  | BehavioralHealthQuestionProperties;
export type ClickEventProperties =
  | CommonAnalyticsProperties
  | SkillsHomeEventProperties
  | BehavioralHealthEventProperties
  | BehavioralHealthFollowUpEventProperties
  | ContentPageProperties
  | ActionCardAnalyticsProperties
  | DiscoverSearchProperties
  | DiscoverSearchPageProperties;
export type ViewEventProperties =
  | CommonAnalyticsProperties
  | BehavioralHealthScreenerCommonProperties
  | BehavioralHealthFollowUpEventProperties
  | ActionCardAnalyticsProperties
  | DiscoverSearchProperties
  | DiscoverSearchPageProperties;

export interface VirtaAnalyticsClient {
  logClickEvent: (event: string, properties: ClickEventProperties) => void;
  logViewEvent: (event: string, properties: ViewEventProperties) => void;
  logCustomEvent?: (event: string, properties: Record<string, unknown>) => void;
}
