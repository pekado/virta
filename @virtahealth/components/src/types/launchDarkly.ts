/*
  LaunchDarkly Client needs to be Project-aware to be able to support querying about
  multiple user bases. For example, the Spark and Patient projects.
*/
export interface LaunchDarklyClient {
  setup(environment: string): Promise<void>;
  identify(
    key: string,
    custom?: Record<string, any>,
    project?: string
  ): Promise<void>;
  getAllFlags<FlagSet>(project?: string): Promise<FlagSet>;
  getJSONFlag<T = object>(
    flag: string,
    defaultReturnValue: Record<string, any>,
    project?: string
  ): Promise<T>;
  getStringFlag(
    flag: string,
    defaultReturnValue: string,
    project?: string
  ): Promise<string>;
  getBooleanFlag(
    flag: string,
    defaultReturnValue: boolean,
    project?: string
  ): Promise<boolean>;
  getFloatFlag(
    flag: string,
    defaultReturnValue: number,
    project?: string
  ): Promise<number>;
  getIntFlag(
    flag: string,
    defaultReturnValue: number,
    project?: string
  ): Promise<number>;
}
