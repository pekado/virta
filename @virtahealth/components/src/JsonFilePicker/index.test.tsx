import * as React from "react";
import { createComponentWithVirtaContext } from "../test";
import { AcceptedFileTypes } from "./types";
import { JsonFilePicker } from "./index";

jest.mock("./Uploader", () => ({
  __esModule: true,
  default: "Uploader",
}));

it("renders the JsonFilePicker", () => {
  const jsonFilePickerContainer = createComponentWithVirtaContext(
    <JsonFilePicker
      accept={[AcceptedFileTypes.pdf]}
      setEncodedFiles={() => undefined}
    />
  );
  expect(jsonFilePickerContainer).toMatchSnapshot();
});
