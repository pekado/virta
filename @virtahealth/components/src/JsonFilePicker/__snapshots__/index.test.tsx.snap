// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the JsonFilePicker 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "black",
        "borderStyle": "solid",
        "borderWidth": 2,
      },
    ]
  }
>
  <Uploader
    accept={
      Array [
        ".pdf",
      ]
    }
    onSetFiles={[Function]}
    testID="file-uploader"
  />
</View>
`;
